gst/edge/edge_src.c

```diff
@@ -0,0 +1,476 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2022 Samsung Electronics Co., Ltd.
+ *
+ * @file    edge_src.c
+ * @date    02 Aug 2022
+ * @brief   Subscribe and push incoming data to the GStreamer pipeline
+ * @author  Yechan Choi <yechan9.choi@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ *
+ */
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "edge_src.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_edgesrc_debug);
+#define GST_CAT_DEFAULT gst_edgesrc_debug
+
+/**
+ * @brief the capabilities of the outputs 
+ */
+static GstStaticPadTemplate srctemplate = GST_STATIC_PAD_TEMPLATE ("src",
+    GST_PAD_SRC, GST_PAD_ALWAYS, GST_STATIC_CAPS_ANY);
+
+/**
+ * @brief edgesrc properties
+ */
+enum
+{
+  PROP_0,
+
+  PROP_HOST,
+  PROP_PORT,
+
+  PROP_LAST
+};
+
+#define gst_edgesrc_parent_class parent_class
+G_DEFINE_TYPE (GstEdgeSrc, gst_edgesrc, GST_TYPE_BASE_SRC);
+
+static void gst_edgesrc_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_edgesrc_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+static void gst_edgesrc_class_finalize (GObject * object);
+
+static GstStateChangeReturn gst_edgesrc_change_state (GstElement * element,
+    GstStateChange transition);
+
+static gboolean gst_edgesrc_start (GstBaseSrc * basesrc);
+static gboolean gst_edgesrc_stop (GstBaseSrc * basesrc);
+static GstCaps *gst_edgesrc_get_caps (GstBaseSrc * basesrc, GstCaps * filter);
+static void gst_edgesrc_get_times (GstBaseSrc * basesrc, GstBuffer * buffer,
+    GstClockTime * start, GstClockTime * end);
+static gboolean gst_edgesrc_is_seekable (GstBaseSrc * basesrc);
+static GstFlowReturn gst_edgesrc_create (GstBaseSrc * basesrc, guint64 offset,
+    guint size, GstBuffer ** out_buf);
+static gboolean gst_edgesrc_query (GstBaseSrc * basesrc, GstQuery * query);
+
+static gchar *gst_edgesrc_get_host (GstEdgeSrc * self);
+static void gst_edgesrc_set_host (GstEdgeSrc * self, const gchar * host);
+
+static guint16 gst_edgesrc_get_port (GstEdgeSrc * self);
+static void gst_edgesrc_set_port (GstEdgeSrc * self, const guint16 port);
+
+/**
+ * @brief initialize the class
+ */
+static void
+gst_edgesrc_class_init (GstEdgeSrcClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+  GstElementClass *gstelement_class = GST_ELEMENT_CLASS (klass);
+  GstBaseSrcClass *gstbasesrc_class = GST_BASE_SRC_CLASS (klass);
+
+  gobject_class->set_property = gst_edgesrc_set_property;
+  gobject_class->get_property = gst_edgesrc_get_property;
+  gobject_class->finalize = gst_edgesrc_class_finalize;
+
+  g_object_class_install_property (gobject_class, PROP_HOST,
+      g_param_spec_string ("host", "Host", "The hostname to send", DEFAULT_HOST,
```

Comment:
Please update the description.
```suggestion
      g_param_spec_string ("host", "Host", "A host address to receive the packets from edgesink", DEFAULT_HOST,
```

Reply:
Updated other descriptions, too.